tools/hipify-perl

```diff
@@ -1161,9 +1167,15 @@ sub simpleSubstitutions {
     $ft{'numeric_literal'} += s/\bCUBLAS_STATUS_NOT_INITIALIZED\b/HIPBLAS_STATUS_NOT_INITIALIZED/g;
     $ft{'numeric_literal'} += s/\bCUBLAS_STATUS_NOT_SUPPORTED\b/HIPBLAS_STATUS_NOT_SUPPORTED/g;
     $ft{'numeric_literal'} += s/\bCUBLAS_STATUS_SUCCESS\b/HIPBLAS_STATUS_SUCCESS/g;
-    $ft{'numeric_literal'} += s/\bCUDA_C_16F\b/HIPBLAS_C_16F/g;
-    $ft{'numeric_literal'} += s/\bCUDA_C_32F\b/HIPBLAS_C_32F/g;
-    $ft{'numeric_literal'} += s/\bCUDA_C_64F\b/HIPBLAS_C_64F/g;
+    if ($rocm_version >= 70000){
+        $ft{'numeric_literal'} += s/\bCUDA_C_16F\b/HIP_C_16F/g;
+        $ft{'numeric_literal'} += s/\bCUDA_C_32F\b/HIP_C_32F/g;
+        $ft{'numeric_literal'} += s/\bCUDA_C_64F\b/HIP_C_64F/g;
+    } else {
+        $ft{'numeric_literal'} += s/\bCUDA_C_16F\b/HIPBLAS_C_16F/g;
+        $ft{'numeric_literal'} += s/\bCUDA_C_32F\b/HIPBLAS_C_32F/g;
+        $ft{'numeric_literal'} += s/\bCUDA_C_64F\b/HIPBLAS_C_64F/g;
+    }
```

Comment:
The value of `HIPBLAS_C_16F` changed in ROCm 6.0, so unfortunately it is not possible to just use `HIP_C_16F` with all versions of hipBLAS. On the bright side, it looks like the removal of the compatibility macros was reverted. https://github.com/ROCm/hipBLAS/commit/585250158ac39ec5052a799666d3ddd0da25233f

Reply:
@cgmb I'm not sure I follow. Is there a minimum ROCm version from which we can just use `HIP_C_16F`?

Reply:
Use of `HIP_C_16F`will be active for ROCm >= v7.0.0, which include hipblas version 3.0.0v.

Reply:
BTW, I hope at some point to use hipify-perl from ROCm, instead of having a slightly-customized version embedded in MAGMA. Customizing a script that literally says "do not change" is obviously not good for compatibility.

Reply:
> Use of `HIP_C_16F`will be active for ROCm >= v7.0.0, which include hipblas version 3.0.0v.

As of ROCm 6.0, `HIPBLAS_C_64F` is just an alias for `HIP_C_16F`, so it's already usable with the HIPBLAS_V2 API. The change that was proposed for ROCm 7.0 (and later backed out) was removing the `HIPBLAS_C_64F` alias, which would _force_ the switch.